### PR TITLE
Initial audio support

### PR DESCRIPTION
**Pull request by hexkyz**
Basic audio output support via "audout:u" service.

### CODE REVIEW

File: nx/source/services/audout.c [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    // Setup the default device
+    if (R_SUCCEEDED(rc))
+    {
+        u32 SampleRate = 0;
+        u32 ChannelCount = 0;
+        PcmFormat Format = 0;
+        AudioOutState State = 0;
+        
+        // Passing an empty device name will open the default "DeviceOut"
+        char DeviceNameIn[DEVICE_NAME_LENGTH];
```
Comment (fincs) [inline nx/source/services/audout.c]: This is uninitialized... perhaps it should have ` = {0}` at the end?

Comment (hexkyz) [reply nx/source/services/audout.c]: Indeed it should

File: nx/include/switch/services/audout.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    PcmFormat_ADPCM = 6,
+} PcmFormat;
+
+typedef enum {
+    AudioOutState_Started = 0,
+    AudioOutState_Stopped = 1,
+} AudioOutState;
+
+/// audio output buffer format
+typedef struct AudioOutBuffer AudioOutBuffer;
```
[the remark this answers is not in the document]
Comment (hexkyz) [reply nx/include/switch/services/audout.h]: Ok. By the way, I based this on applet.h which appears to suffer from the same issue.

[the remark this answers is not in the document]
Comment (fincs) [reply nx/include/switch/services/audout.h]: Actually nvm, looks like doxygen can handle it.

**Pull request merged**